egs/libri_css/asr1_mono/local/decode.sh

```diff
@@ -0,0 +1,191 @@
+nbpe=5000
+bpemode=unigram
+train_set=train_960
+dict=data/lang_char/${train_set}_${bpemode}${nbpe}_units.txt
+bpemodel=data/lang_char/${train_set}_${bpemode}${nbpe}
+
+. ./utils/parse_options.sh
+
+. ./cmd.sh
+. ./path.sh
+
+$use_oracle_segments && [ $stage -le 6 ] && stage=6
+
+## Ensure that webrtcvad is installed
+if [ -z `pip freeze | grep webrtcvad` ]; then
+  pip install webrtcvad
+fi
+
+
+
+#######################################################################
+# Perform SAD on the dev/eval data using py-webrtcvad package
+#######################################################################
+
+if [ $stage -le 1 ]; then
+  for datadir in ${test_sets}; do
+    test_set=data/${datadir}
+    if [ ! -f ${test_set}/wav.scp ]; then
+      echo "$0: Not performing SAD on ${test_set}"
+      exit 0
+    fi
+
+    # Perform segmentation
+    local/segmentation/apply_webrtcvad.py --mode 0 $test_set | sort > $test_set/segments
+
+    # Create dummy utt2spk file from obtained segments
+    awk '{print $1, $2}' ${test_set}/segments > ${test_set}/utt2spk
+    utils/utt2spk_to_spk2utt.pl ${test_set}/utt2spk > ${test_set}/spk2utt
+
+    # Generate RTTM file from segmentation performed by SAD. This can
+    # be used to evaluate the performance of the SAD as an intermediate
+    # step.
+    steps/segmentation/convert_utt2spk_and_segments_to_rttm.py \
+      ${test_set}/utt2spk ${test_set}/segments ${test_set}/rttm
+
+    if [ $score_sad == "true" ]; then
+      echo "Scoring $datadir.."
+      # We first generate the reference RTTM from the backed up utt2spk and segments
+      # files.
+      ref_rttm=${test_set}/ref_rttm
+      steps/segmentation/convert_utt2spk_and_segments_to_rttm.py ${test_set}/utt2spk.bak \
+        ${test_set}/segments.bak ${test_set}/ref_rttm
+
+      md-eval.pl -r $ref_rttm -s ${test_set}/rttm |\
+        awk 'or(/MISSED SPEECH/,/FALARM SPEECH/)'
+    fi
+  done
+fi
+
+#######################################################################
+# Feature extraction for the dev and eval data
+#######################################################################
+if [ $stage -le 2 ]; then
+  # mfccdir should be some place with a largish disk where you
+  # want to store MFCC features.
+  mfccdir=mfcc
+  for x in ${test_sets}; do
+    steps/make_mfcc.sh --nj $nj --cmd "$train_cmd" \
+      --mfcc-config conf/mfcc_hires.conf \
+      data/$x exp/make_mfcc/$x $mfccdir
+  done
+fi
+
+#######################################################################
+# Perform diarization on the dev/eval data
+#######################################################################
+if [ $stage -le 3 ]; then
+  for datadir in ${test_sets}; do
+    ref_rttm=data/${datadir}/ref_rttm
+    steps/segmentation/convert_utt2spk_and_segments_to_rttm.py data/${datadir}/utt2spk.bak \
+      data/${datadir}/segments.bak $ref_rttm
+    diar_nj=$(wc -l < "data/$datadir/wav.scp") # This is important especially for VB-HMM
+
```

Comment:
There seems to be a VB-HMM script "diarize_bhmm.sh" in this PR which is not used. Can this be used as an alternative to "diarize_spectral.sh"? Can you please pass the diarization method from run.sh to here in that case.

Reply:
Done.